### ISSUE TITLE
Renderbase64 doesn't always honor viewport height

### ISSUE THREAD

**Issue by jeantil** [ariya/phantomjs#12875]
I am unsure how to test this directly but here are the steps to reproduce the problem through webdriver for both phantomjs 1.9.8 and phantomjs2.0@05d7e3e : 

```
SESSIONID=`curl -s -X POST -d '{"requiredCapabilities":{},"desiredCapabilities":{}}' localhost:8000/session | jq -r '.sessionId'`
curl -s -X POST -d '{"url":"https://openoox.com"}' localhost:8000/session/$SESSIONID/url | jq '.'
curl -s -X POST -d '{"width":1280,"height":768}' localhost:8000/session/$SESSIONID/window/current/size | jq '.'
curl -s -X GET localhost:8000/session/$SESSIONID/window/current/size | jq '.'
#{
#  "sessionId": "50cee280-9b42-11e4-9a9d-d75c0f3f8c5a",
#  "status": 0,
#  "value": {
#    "height": 768,
#    "width": 1280
#  }
#}
curl -s -X GET localhost:8000/session/$SESSIONID/screenshot | jq -r '.value' | base64 -D >/tmp/screenshot.png

file /tmp/screenshot.png
# /tmp/screenshot.png: PNG image data, 1280 x 1333, 8-bit/color RGBA, non-interlaced
```

As you can see the window size is set ( https://github.com/ariya/phantomjs/blob/48fabe06463460d2fb7026d6df9783216e26265c/src/ghostdriver/request_handlers/session_request_handler.js#L285 ) and reported  by `viewportSize` ( https://github.com/ariya/phantomjs/blob/48fabe06463460d2fb7026d6df9783216e26265c/src/ghostdriver/request_handlers/session_request_handler.js#L294 )  to be 

``` json
 {
   "height": 768,
    "width": 1280
 }
```

yet the screenshot rendered by  base64 ( https://github.com/ariya/phantomjs/blob/48fabe06463460d2fb7026d6df9783216e26265c/src/ghostdriver/request_handlers/session_request_handler.js#L468 ) ends up being 1280 x 1333. 

failing websites : 
- https://openoox.com
- http://lemonde.fr
- https://facebook.com
- https://plus.google.com
- https://github.com
- https://yahoo.com
- https://flickr.com

working websites :
- https://google.com
- https://pinterest.com 
- https://twitter.com

**Comment by jeantil** [comment]
This is linked to #10217 which is considered fixed by d844d540.  This will cause render to only consider the size of the content of the window (which will be influenced by viewport width but not by its height). 

Digging some more, it seems there is a clipRect method available in the phantomJs API which would enforce the viewport size when  rendering the screenshot. 
This is not used by the embedded webdriver implementation when getting a screenshot. Unfortunately,  the webdriver [JsonWireProtocol](https://code.google.com/p/selenium/wiki/JsonWireProtocol) doesn't offer an option which could be used to distinguish wether to clip or not when taking a screenshot.  

I have opened https://code.google.com/p/selenium/issues/detail?id=8384 to request an enhancement to the wire protocol to help decide whether to clip or not to clip.

**Comment by jeantil** [comment]
it seems the situation is evolving on this, the json  wire protocol is now a w3c project and they are currently changing the takeScreenshot specification to capture viewport instead of full page (see 
https://github.com/w3c/webdriver/pull/98 and https://github.com/SeleniumHQ/selenium/issues/646#issuecomment-111043913)
